src/openzaak/components/documenten/api/filters.py

```diff
@@ -30,13 +33,119 @@
 logger = logging.getLogger(__name__)
 
 
+class ZaakFilter(FkOrUrlFieldFilter):
+    def filter(self, qs, value):
+        if not value:
+            return qs
+
+        if not check_path(value, "zaken"):
+            logger.debug(
+                "URL (%s) is not a zaak, filtering to empty result set.", value
+            )
+            return qs.none()
+
+        return super().filter(qs, value)
+
+
 class EnkelvoudigInformatieObjectListFilter(FilterSet):
-    trefwoorden = CharArrayFilter(field_name="trefwoorden", lookup_expr="contains")
+    creatiedatum__gte = filters.DateFilter(
+        help_text=mark_experimental(
+            "De aanmakings datum van dit informatie object (groter of gelijk aan de gegeven datum)."
+        ),
+        field_name="creatiedatum",
+        lookup_expr="gte",
+    )
+    creatiedatum__lte = filters.DateFilter(
+        help_text=mark_experimental(
+            "De aanmakings datum van dit informatie object (kleiner of gelijk aan de gegeven datum)."
+        ),
+        field_name="creatiedatum",
+        lookup_expr="lte",
+    )
+    auteur = filters.CharFilter(
+        help_text=mark_experimental(
+            "De persoon of organisatie die dit informatie object heeft aangemaakt."
+        ),
+        lookup_expr="icontains",
+    )
+    beschrijving = filters.CharFilter(
+        help_text=mark_experimental("De bescrijving van dit informatie object."),
+        lookup_expr="icontains",
+    )
+    locked = filters.BooleanFilter(
+        help_text=mark_experimental(
+            "De indicatie of dit informatie object gelocked is of niet."
+        ),
+        method="locked_filter",
+    )
+    titel = filters.CharFilter(
+        help_text=mark_experimental("De titel van het informatie object."),
+        lookup_expr="icontains",
+    )
+    trefwoorden = CharArrayFilter(
+        help_text=mark_experimental(
+            "Filter op de trefwoorden die eventueel gelinkt zijn aan het informatie object."
+        ),
+        lookup_expr="overlap",
+    )
+    vertrouwelijkheidaanduiding = filters.ChoiceFilter(
+        help_text=mark_experimental(
+            "De vertrouwelijkheidaanduiding van het informatie object"
+        ),
+        choices=VertrouwelijkheidsAanduiding.choices,
+    )
+    object_type = filters.ChoiceFilter(
+        help_text=mark_experimental("Het type van het gerelateerde OBJECT."),
+        field_name="canonical__objectinformatieobject__object_type",
+        lookup_expr="exact",
+        choices=ObjectInformatieObjectTypes.choices,
+    )
+
+    zaak = ZaakFilter(
+        help_text=mark_experimental(
+            "URL-referentie naar de gerelateerde ZAAK (in deze of een andere API)."
+        ),
+        queryset=EnkelvoudigInformatieObject.objects.all(),
+        field_name="canonical__objectinformatieobject__zaak",
+    )
+
+    ordering = OrderingFilter(
+        help_text=mark_experimental("Sorteer op."),
+        fields=(
```

Comment:
I looked over this in the first review, but in the PR description I see `bestandstype` as an ordering filter as well, which is missing. It's not a field on any of the models though, I think that might be a filter on `formaat`? @joeribekker